pkg/utils/plugin/plugin.go

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+package plugin
+
+import (
+	"sync"
+
+	"github.com/azure/kaito/pkg/inference"
+)
+
+type Registration struct {
+	Name     string
+	Instance inference.Model
+}
+
+type ModelRegister struct {
+	sync.RWMutex
+	models map[string]*Registration
+}
+
+var KaitoModelRegister ModelRegister
+
+// Register allows model to be added
+func (reg *ModelRegister) Register(r *Registration) {
+	reg.Lock()
+	defer reg.Unlock()
+	if r.Name == "" {
+		panic("model name is not specified")
+	}
+
+	if reg.models == nil {
+		reg.models = make(map[string]*Registration)
+	}
+
+	reg.models[r.Name] = r
+}
+
+func (reg *ModelRegister) MustGet(name string) inference.Model {
+	reg.Lock()
+	defer reg.Unlock()
+	if _, ok := reg.models[name]; ok {
+		return reg.models[name].Instance
+	}
+	panic("model is not registered")
```

Comment:
should we panic? maybe we just return error here

Reply:
With "Must", we panic.